cdci_data_analysis/analysis/parameters.py

```diff
@@ -440,9 +450,25 @@ def __init__(self,value=None,T_format=None,name=None,Time_format_name=None):
                                   allowed_units=None)
                                   #wtform_dict=wtform_dict)
 
-
         self._set_time(value,format=T_format)
 
+    @staticmethod
+    def set_group_par(par_group, par_dic, params_not_to_be_included=None, verbose=False):
+        if params_not_to_be_included is None:
+            params_not_to_be_included = []
+        num_time_params = 0
+        for par in par_group:
+            if par.name is not None and par.name not in params_not_to_be_included:
+                par.set_from_form(par_dic, verbose=verbose)
+                num_time_params += 1
+        if num_time_params > 0:
+            par_dic['T_format'] = 'isot'
```

Comment:
this could be more generic, apply to all parameters. `'T_format'` is in `units_name` actually. And the default format should be stored somewhere too.

Then the decision to add a request parameter with default units name can be made more generally. Also preferrably by a function returning complementary parameter dict, not by modifying passed dict.

Reply:
the problem in this case, for a `Time` parameter more specifically, is that there is one single units name (`T_format`) that will be applied to all the time parameters (`T1` and `T2`), so it becomes tricky to generalize

Reply:
should it be really a problem? Each parameter could take care of verifying and possibly setting it's units field. If many parameters do it for the same units field, the could behave in a way which safely stacks these validations/sets.

* if unit field is set (either by input or by another parameter before): check that the value fits it
* if unit field is not set (so not in input and no other parameters of the same type considered before): set the default, validate

Reply:
what we want is that the unit field is always set to `isot` (or any other default), no matter what, that is why I implemented this approach only for Time parameters

after setting the value (in the form) to the `isot` format for all the Time parameters, set also the units name once for all

Reply:
does it make sense?

Reply:
all parameters with unit should be converted to the same, default, unit. Else the same problem as here will occur. Essentially job_id and the URL should not depend on particular representation of the parameter.

Reply:
> all parameters with unit should be converted to the same, default, unit. Else the same problem as here will occur. Essentially job_id and the URL should not depend on particular representation of the parameter.

ok, this could also help in avoid using the `set_group_par` function

Reply:
> > all parameters with unit should be converted to the same, default, unit. Else the same problem as here will occur. Essentially job_id and the URL should not depend on particular representation of the parameter.
> 
> ok, this could also help in avoid using the `set_group_par` function

Just wanted to clarify that it applies to all parameters which have meaningful units. But it also can be things like different number representations.

Reply:
Till now I have seen only applied to `Time` parameters. But in the future there could be also other parameters to be treated in the same way

Reply:
we should try if the same already happens for, e.g., E1_keV, with values like 25 or 25.0 . Does it get normalized somewhere?

Reply:
For `E1_keV` (or `E2_keV`), the value that is passed (which will always be a string) is checked if it is in a valid format (`int` of `float` are accepted), so both the options you mentioned are accepted.

Regarding the setting of the `units_name` entry in the form, this is now applied for each parameter where this is specified, and to the defined `default_units`.

Reply:
> For `E1_keV` (or `E2_keV`), the value that is passed (which will always be a string) is checked if it is in a valid format (`int` of `float` are accepted), so both the options you mentioned are accepted.
> 
> Regarding the setting of the `units_name` entry in the form, this is now applied for each parameter where this is specified, and to the defined `default_units`.

checked - yes, but will it be transformed to the same form before generation of `job_id` and the URL? 25 and 25.0 mean exactly the same here but might look different.

Reply:
passing 20 is different from passing 20.0 (or even 20.): the first will be interpreted as an integer whereas the second as a float, and will also yield different `job_id`

Reply:
To address this, we should apply the approach used for the `Time` parameter, also to the `SpectralBoundary`, and perhaps also to other parameters if necessary. But in a separate dedicated PR.

Reply:
Maybe separate PR, yes, but what are the consequences of what is done in this one? Is it not so that at the general level (not specific to time), parameter value is taken from the instance, after it's set. In this case, no matter what is the type/unit, it will be valid. 

Maybe this general approach is suitable for time too? If so, it would be in this PR. And it will spread on other cases automatically.

Reply:
What I meant is that for each type of parameter, a default format of the value should be adopted, and then used (ie set inside `par_dic`, and used for later `job_id`(s) generation). Therefore for each parameter a different discussion (and related PR) should be opened.

As for this PR, now the `job_id` calculation will always be performed with the time values in `isot` format, and the `units_name` set to `isot`.

Reply:
> What I meant is that for each type of parameter, a default format of the value should be agreed, and then adopted (ie set inside `par_dic`, and used for later `job_id`(s) generation). Therefore for each parameter a different discussion (and related PR) should be opened.
> 
> As for this PR, now the `job_id` calculation will always be performed with the time values in `isot` format, and the `units_name` set to `isot`.

That's a lot of nearly-identical PR's for each parameter. There can be a generic adjustment of Parameter class with reasonable defaults. Since some generic adjustments will be necessary, there is no sense to separate in many.

But if you want to be done with this block, we can merge this, and develop a suitable treatment in one another PR.

Reply:
Yes, in a dedicated PR. This was specific to `Time` parameter.

Reply:
> Yes, in a dedicated PR. This was specific to `Time` parameter.

For something specitifc to `Time` parameter this PR made lot's of changes in generic part, not directly related to `Time` parameter.